app/models/ability.rb

```diff
@@ -0,0 +1,20 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    if user.superadmin?
+        can :manage, :all
+    elsif user.admin?
+        can [:read, :create], Message
+        can [:update,:destroy], Message, :category => "Public"
+    elsif user.user?
+        can :create, Message, :category => "Private"
+        can :read, Message
+    elsif user.guest?
+        can :read, Message, :category => "Public"
+        # can :read, Message do |message|
```

Comment:
Remove commented code. You can always refer to it via blame (https://help.github.com/articles/using-git-blame-to-trace-changes-in-a-file/)